Zephyr doesn't build if x86_64 SDK toolchain isn't install

If using the Zephyr SDK and only having one arch toolchain (for example ARM) installed on a x86-host system we get the following:

```
root@9b2eebb80c9e:~/zephyr/samples/hello_world/build# cmake -DBOARD=frdm_k64f ..
-- Zephyr version: 2.1.99
-- Found PythonInterp: /usr/bin/python3.6 (found suitable version "3.6.9", minimum required is "3.6") 
-- Selected BOARD frdm_k64f
-- Found west: /usr/local/bin/west (found suitable version "0.7.0", minimum required is "0.6.0")
CMake Error at /root/zephyr/cmake/compiler/gcc/generic.cmake:8 (message):
  Zephyr was unable to find the toolchain.  Is the environment misconfigured?

  User-configuration:

  ZEPHYR_TOOLCHAIN_VARIANT: zephyr

  Internal variables:

  CROSS_COMPILE: /root/zephyr-sdk/x86_64-zephyr-elf/bin/x86_64-zephyr-elf-

  TOOLCHAIN_HOME: /root/zephyr-sdk

Call Stack (most recent call first):
  /root/zephyr/cmake/generic_toolchain.cmake:70 (include)
  /root/zephyr/cmake/app/boilerplate.cmake:462 (include)
  CMakeLists.txt:5 (include)


-- Configuring incomplete, errors occurred!
```

--- a/cmake/toolchain/zephyr/0.11/generic.cmake
+++ b/cmake/toolchain/zephyr/0.11/generic.cmake
@@ -6,9 +6,29 @@ set(COMPILER gcc)
 set(LINKER ld)
 set(BINTOOLS gnu)
 
-set(CROSS_COMPILE_TARGET x86_64-${TOOLCHAIN_VENDOR}-elf)
-set(SYSROOT_TARGET       x86_64-${TOOLCHAIN_VENDOR}-elf)
+# Find some toolchain that is distributed with this particular SDK
 
-set(CROSS_COMPILE ${TOOLCHAIN_HOME}/${CROSS_COMPILE_TARGET}/bin/${CROSS_COMPILE_TARGET}-)
-set(SYSROOT_DIR ${ZEPHYR_SDK_INSTALL_DIR}/sysroots/${SYSROOT_TARGET}/usr)
+file(GLOB toolchain_paths
+  LIST_DIRECTORIES true
+  ${TOOLCHAIN_HOME}/xtensa/*/*-zephyr-elf
+  ${TOOLCHAIN_HOME}/*-zephyr-elf
+  ${TOOLCHAIN_HOME}/*-zephyr-eabi
+  )
+
+if(toolchain_paths)
+  list(GET toolchain_paths 0 some_toolchain_path)
+
+  get_filename_component(one_toolchain_root "${some_toolchain_path}" DIRECTORY)
+  get_filename_component(one_toolchain "${some_toolchain_path}" NAME)
+
+  set(CROSS_COMPILE_TARGET ${one_toolchain})
+  set(SYSROOT_TARGET	   ${one_toolchain})
+endif()
+
+if(NOT CROSS_COMPILE_TARGET)
+  message(FATAL_ERROR "Unable to find 'x86_64-zephyr-elf' or any other architecture in ${TOOLCHAIN_HOME}")
+endif()
+
+set(CROSS_COMPILE ${one_toolchain_root}/${CROSS_COMPILE_TARGET}/bin/${CROSS_COMPILE_TARGET}-)
+set(SYSROOT_DIR   ${one_toolchain_root}/${SYSROOT_TARGET}/${SYSROOT_TARGET})
 set(TOOLCHAIN_HAS_NEWLIB ON CACHE BOOL "True if toolchain supports newlib")

